Subject: [PATCH] Remove the unnecessary window animation request

When AppWindowAnimator set with dummy animation, there is no need to
request window animation

When the device's performance is low (CPU run with one Core and in low frequency),
the dummy animation may hold the WMS synchornized lock cycling,
it will drag down the performance

diff --git a/services/core/java/com/android/server/wm/WindowAnimator.java b/services/core/java/com/android/server/wm/WindowAnimator.java
index 85a9624..7ca7342 100644
--- a/services/core/java/com/android/server/wm/WindowAnimator.java
+++ b/services/core/java/com/android/server/wm/WindowAnimator.java
@@ -604,7 +604,7 @@ private void updateWallpaperLocked(int displayId) {
             // If this window is animating, make a note that we have
             // an animating window and take care of a request to run
             // a detached wallpaper animation.
-            if (winAnimator.mAnimating) {
+            if (winAnimator.mAnimating && !winAnimator.isDummyAnimation()) {
                 if (winAnimator.mAnimation != null) {
                     if ((flags & FLAG_SHOW_WALLPAPER) != 0
                             && winAnimator.mAnimation.getDetachWallpaper()) {
